File: tests/managed_integration/fixtures/lua/hello.lua
-- Tier 1: Basic Lua hello-world. Tests print, local variables,
-- string concatenation, and basic math operations.
local name = "World"
print("Hello, " .. name .. "!")
print("Lua fixture for RetDec.")

local function factorial(n)
    if n <= 1 then return 1 end
    return n * factorial(n - 1)
end

print("5! = " .. factorial(5))
print("10! = " .. factorial(10))

File: tests/managed_integration/fixtures/csharp/Hello.cs
// Tier 1: Basic C# hello-world. Tests namespace, class, Main entry point,
// Console.WriteLine, and string interpolation.
using System;

namespace RetDecTest
{
    class Hello
    {
        static void Main(string[] args)
        {
            string name = args.Length > 0 ? args[0] : "World";
            Console.WriteLine($"Hello, {name}!");
            Console.WriteLine("C# fixture for RetDec.");
        }
    }
}
